src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactory.java

```diff
@@ -1026,9 +1263,70 @@ private static GencodeFuncotation createIntronFuncotation(final VariantContext v
             );
         }
 
+        // Set our version:
+        gencodeFuncotationBuilder.setVersion(version);
+
         return gencodeFuncotationBuilder.build();
     }
 
+    /**
+     * Get the bases around the given variant (as specified by {@code refAllele} and {@code altAllele}).
+     * The number of bases before and after the variant is specified by {@link #referenceWindow}.
+     * @param refAllele The reference {@link Allele} for the variant.
+     * @param altAllele The alternate {@link Allele} for the variant.
+     * @param reference The {@link ReferenceContext} for the variant, with the current window around the variant.
+     * @param strand The {@link Strand} on which the variant occurs.
+     * @return A {@link String} of bases of length {@link #referenceWindow} * 2 + |variant|
+     */
+    private static String getReferenceBases(final Allele refAllele, final Allele altAllele, final ReferenceContext reference, final Strand strand ) {
+
+        // Adjustment for the start of the variant based on whether it's a deletion or not.
+        // NOTE: This is an artifact of the GATK representation of variants (that they must be at least 1 base)
+        final int deletionStartAdjustment =  GATKVariantContextUtils.isDeletion(refAllele, altAllele) ? 1 : 0;
+
+        if ( strand == Strand.POSITIVE ) {
```

Comment:
I'm not sure why you need the strand to figure out the reference window to use around a variant.  Can you explain?

Reply:
I use the reference window to create the in-frame alleles so I need to make it strand-correct.  When I report it in the funcotation itself, I make sure it's always in the forward direction.